haiku-apps/flyingtroll/flyingtroll-1.0.recipe

```diff
@@ -0,0 +1,43 @@
+SUMMARY="A simple app for browsing local documentation"
+DESCRIPTION="FlyingTroll can be used for browsing local documentation. \
+It has its own HTML layout engine. There's absolutely no JS, CSS, nor \
+network support. But it has a lot of potential."
+HOMEPAGE="https://github.com/HaikuArchives/FlyingTroll"
+COPYRIGHT="2007 Nikita Zalyotov"
+LICENSE="MIT"
+REVISION="1"
+gitSrcRevision="4781ab9600b4042fafd6e8fe76d6d3947a14d8c1"
+SOURCE_URI="https://github.com/HaikuArchives/FlyingTroll/archive/$gitSrcRevision.tar.gz"
+CHECKSUM_SHA256="f3175716c49d17e668ef6bca136fc03789c08a0c6b5500062533208edd07a405"
+SOURCE_DIR="FlyingTroll-$gitSrcRevision/src"
```

Comment:
I though haikuports did not work correctly when SOURCE_DIR was more than one level, but it does.
Well, I'm quite sure it did _not_ some time ago.
I would recommend a few changes, to make sure all versions of haikuporter would work for this recipe.

I'd also rename the recipe to `flyingtroll-1.0~git.recipe` and insert a SOURCE_FILENAME.
```diff
-SOURCE_DIR="FlyingTroll-$gitSrcRevision/src"
+SOURCE_FILENAME="FlyingTroll-$portVersion-$gitSrcRevision.tar.gz"
+SOURCE_DIR="FlyingTroll-$gitSrcRevision"
```

Reply:
@fbrosson Does adding `~git` do something magical?